src/App/Template/AbstractTemplate.php

```diff
@@ -16,11 +16,88 @@ abstract class AbstractTemplate extends AbstractWebTemplate
      */
     const ASSETS_VERSION = '1.0';
 
+    const CORE_TEMPLATE_ACTIONS = [
+        'siteName',
+        'google',
+        'typekit',
+        'htmlAttrStructure',
+        'htmlAttr',
+        'assetsVersion',
+        'copyright',
+        'copyrightName',
+        'copyrightYear',
+        'criticalCss',
+        'templateName',
+        'title',
+        'currentLocale',
+        'currentUrl',
+        'currentLanguage',
+        'metaTitle',
+        'metaDescription',
+        // 'metaImage',
+        'opengraphType',
+        // 'opengraphImage',
+        'seoMetadata',
+        'baseUrl',
+        'availableLanguages',
+        'hasSeoMetadata',
+        'documentTitle',
+        'hasAlternateTranslations',
+        'alternateTranslations',
+    ];
+
+    /**
+     * Gets the data keys on this entity.
+     * @see
+     *
+     * @return array
+     */
+    public function keys($keys = null)
+    {
+        $methods = $this->retrieveDynamicMethods();
+        return array_merge(self::CORE_TEMPLATE_ACTIONS, $methods);
+    }
+
+    private function retrieveDynamicMethods()
+    {
+        $reflectionClass = new \ReflectionClass($this);
+        $reflectionMethods = $reflectionClass->getMethods();
+        $methodKeys = [];
+        foreach ($reflectionMethods as $reflectionMethod) {
+            if (substr($reflectionMethod->class, 0, 3) !== 'App') {
+               continue;
+            }
+
+            if (substr($reflectionMethod->name, 0, 3) === 'get') {
+               $methodKeys[] = lcfirst(substr($reflectionMethod->name, 3));
+            }
+        }
+        return $methodKeys;
+    }
 
 
     // Site Head/Metatags
     // ============================================================
 
+    /**
+     * Format an alternate translation for the given translatable model.
+     *
+     * Note: The application's locale is already modified and will be reset
+     * after processing all available languages.
+     *
+     * @param  mixed $context      The translated {@see \Charcoal\Model\ModelInterface model}
+     *     or array-accessible structure.
+     * @param  array $localeStruct The currently iterated language.
+     * @return array Returns a link structure.
+     */
+    protected function formatAlternateTranslation($context, array $localeStruct)
```

Comment:
Note that I had to overwrite the translation URL because Twig does not accept object. The formatAlternateTranslationUrl returning a Slim\Http\Uri object in the translator package)

Reply:
Hrm. I think we should ensure `formatAlternateTranslationUrl()` returns a string.